common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -425,8 +429,8 @@ workflow_state = ? ` +
 		`and visibility_ts = ? ` +
 		`and task_id = ?`
 
-	templateUpdateWorkflowExecutionQuery = `UPDATE executions ` +
-		`SET execution = ` + templateWorkflowExecutionType + `, next_event_id = ? ` +
+	templateCheckWorkflowExecutionQuery = `UPDATE executions ` +
```

Comment:
I think we should let the reset go through even if next_event_id changes.  All we need to make sure is current execution record does not change during this operation.

Reply:
Resolved offline. We still need to check next_event_id to prevent corrupting the workflow, there shouldn't be any cost or bad customer experience because of that